tests/e2e-playwright/tests/jupyterlab.py

```diff
@@ -0,0 +1,98 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-statements
+# pylint: disable=unnecessary-lambda
+
+import re
+from http import HTTPStatus
+from typing import Final
+
+from playwright.sync_api import APIRequestContext, Page
+from pydantic import AnyUrl
+from tenacity import Retrying
+from tenacity.retry import retry_if_exception_type
+from tenacity.stop import stop_after_attempt
+from tenacity.wait import wait_fixed
+
+projects_uuid_pattern: Final[re.Pattern] = re.compile(
+    r"/projects/([0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12})"
+)
+
+
+def on_web_socket(ws) -> None:
+    print(f"WebSocket opened: {ws.url}")
+    ws.on("framesent", lambda payload: print(payload))
+    ws.on("framereceived", lambda payload: print(payload))
+    ws.on("close", lambda payload: print("WebSocket closed"))
+
+
+def test_jupyterlab(
+    page: Page,
+    log_in_and_out: None,
+    api_request_context: APIRequestContext,
+    product_url: AnyUrl,
+    product_billable: bool,
+    service_key: str,
+    service_test_id: str,
+):
+    # connect and listen to websocket
+    page.on("websocket", on_web_socket)
+
+    # open services tab and filter for the service
+    page.get_by_test_id("servicesTabBtn").click()
+    _textbox = page.get_by_role("textbox", name="search")
+    _textbox.fill(service_key)
+    _textbox.press("Enter")
+    page.get_by_test_id(service_test_id).click()
+
+    with page.expect_response(projects_uuid_pattern) as response_info:
+        # Project detail view pop-ups shows
+        page.get_by_test_id("openResource").click()
+        if product_billable:
+            # Open project with default resources
+            page.get_by_test_id("openWithResources").click()
+        page.wait_for_timeout(1000)
+
+    # Get project uuid, will be used to delete this project in the end
+    print(f"projects uuid endpoint captured: {response_info.value.url}")
+    match = projects_uuid_pattern.search(response_info.value.url)
+    assert match
+    extracted_uuid = match.group(1)
+
+    # Wait until iframe is shown and create new notebook with print statement
+    page.frame_locator(".qx-main-dark").get_by_role(
+        "button", name="New Launcher (Ctrl+Shift+L)"
+    ).click(timeout=600000)
+    page.frame_locator(".qx-main-dark").locator(".jp-LauncherCard-icon").first.click()
+    _jupyterlab_ui = (
+        page.frame_locator(".qx-main-dark")
+        .get_by_label("Untitled.ipynb")
+        .get_by_role("textbox")
+    )
+    _jupyterlab_ui.fill("print('test')")
```

Comment:
From my point of view not writing something to the ports means that we are not testing enough.
All previous tests always tested this.

I think it's insufficient and wrong. Here you are testing only the jupyter-notebook base functionality (which we already know it works).

Ports are more important than this

Reply:
As already mentioned, port functionality will be tested in separate test focused on port testing. This test is testing only basic functionality of starting a jupyterlab test (the same as we are testing basic functionality of sim4life and it already helped us many times). It is not true at all that we know that it already works, as we are using these tests for monitoring of the platform during our batman slides.

Reply:
here are some of the thoughts: https://github.com/ITISFoundation/osparc-simcore/issues/5306

Reply:
so we are changing the logic of this OK